### PR TITLE
ENH: Merge T2w images and coregister to T1w template

### LINKED ISSUE #69: Register the FLAIR or T2 to the T1w space
Coregister FLAIR and/or T2 into the T1w space, and add this option as derivatives for fmriprep.

### PR DESCRIPTION
**Pull request by effigies**
The approach taken here is to merge the T2w images in exactly the same way as T1w images (N4 -> mri_robust_template) and then bbregister. This is the approach taken by `recon-all`, and seems the safest way to do this without running brain extraction first. Therefore this is only available if FreeSurfer is run.

Closes #69.

We could probably throw in #13 at the same time and pass the merged T2w image to FreeSurfer...

### CODE REVIEW

**Comment by effigies** [comment]
### Raw

image

### Preprocessed

image

File: smriprep/workflows/anatomical.py [hunk 1]
```diff
@@ -522,6 +529,55 @@ def _check_img(img):
...
+                contrast_type="t2",
+                init="coreg",
+                dof=6,
+                out_lta_file=True,
+                args="--gm-proj-abs 2 --wm-proj-abs 1",
+            ),
+            name="bbreg",
+        )
+        coreg_xfms = pe.Node(niu.Merge(2), name="merge_xfms", run_without_submitting=True)
+        t2tot1_xfm = pe.Node(ConcatenateXFMs(), name="t2tot1_xfm", run_without_submitting=True)
```
Comment (mgxd) [inline smriprep/workflows/anatomical.py:551]: very minor, but just to stay consistent with the other nodes
```suggestion
        t2wtot1w_xfm = pe.Node(ConcatenateXFMs(), name="t2wtot1w_xfm", run_without_submitting=True)
```

File: smriprep/workflows/anatomical.py [hunk 2]
```diff
@@ -522,6 +529,55 @@ def _check_img(img):
...
+        workflow.connect([
+            (inputnode, t2w_template_wf, [('t2w', 'inputnode.t1w')]),
+            (t2w_template_wf, bbreg, [('outputnode.t1w_ref', 'source_file')]),
+            (surface_recon_wf, bbreg, [
+                ('outputnode.subject_id', 'subject_id'),
+                ('outputnode.subjects_dir', 'subjects_dir'),
+            ]),
+            (bbreg, coreg_xfms, [('out_lta_file', 'in1')]),
+            (surface_recon_wf, coreg_xfms, [('outputnode.fsnative2t1w_xfm', 'in2')]),
+            (coreg_xfms, t2tot1_xfm, [('out', 'in_xfms')]),
```
Comment (mgxd) [inline smriprep/workflows/anatomical.py:571]:
```suggestion
            (coreg_xfms, t2wtot1w_xfm, [('out', 'in_xfms')]),
```

File: smriprep/workflows/anatomical.py [hunk 3]
```diff
@@ -522,6 +529,55 @@ def _check_img(img):
...
+            (t2w_template_wf, t2w_resample, [('outputnode.t1w_ref', 'input_image')]),
+            (brain_extraction_wf, t2w_resample, [
+                (('outputnode.bias_corrected', _pop), 'reference_image'),
+            ]),
+            (t2tot1_xfm, t2w_resample, [('out_xfm', 'transforms')]),
```
Comment (mgxd) [inline smriprep/workflows/anatomical.py:576]:
```suggestion
            (t2wtot1w_xfm, t2w_resample, [('out_xfm', 'transforms')]),
```

**Review by mgxd — APPROVED** [review]
Looks good, now that we are double dipping with some workflows, it may be time to change the parameters/inputs/outputs to reflect this generalization

**Comment by effigies** [comment]
Tests verifying that this doesn't break for no T2w images, and I've rerun locally with T2 and gotten the same result. Merging.

**Pull request merged**